### PR TITLE
test(cli): add test for verify-plugin plugin-helpers

### PR DESCRIPTION
**Pull request by fossamagna**
#### Description of changes

Increase test coverage of amplify-cli/plugin-helpers.

#### Description of how you validated changes

lerna --scope @aws-amplify/cli run test

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-cli/src/__tests__/plugin-helpers/verify-plugin.test.ts [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+        type: 'util',
+        commands: ['version', 'help'],
+        eventHandlers: [AmplifyEvent.PreInit],
+      };
+      readJsonMock.mockReturnValueOnce(amplifyPluginJson);
+
+      // read core package.json
+      readJsonMock.mockReturnValueOnce(corePluginJson);
+
+      const result = await verifyPlugin(path.join(__dirname, '../../../__mocks__/invalid-plugin'));
```
Comment (cjihrig) [inline packages/amplify-cli/src/__tests__/plugin-helpers/verify-plugin.test.ts:237]:
```suggestion
      const result = await verifyPlugin(path.join(__dirname, '..', '..', '..', '__mocks__', 'invalid-plugin'));
```

File: packages/amplify-cli/src/__tests__/plugin-helpers/verify-plugin.test.ts [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+        type: 'util',
+        commands: ['version', 'help'],
+        eventHandlers: [AmplifyEvent.PreInit],
+      };
+      readJsonMock.mockReturnValueOnce(amplifyPluginJson);
+
+      // read core package.json
+      readJsonMock.mockReturnValueOnce(corePluginJson);
+
+      const result = await verifyPlugin(path.join(__dirname, '../../../__mocks__/valid-plugin'));
```
Comment (cjihrig) [inline packages/amplify-cli/src/__tests__/plugin-helpers/verify-plugin.test.ts:280]:
```suggestion
      const result = await verifyPlugin(path.join(__dirname, '..', '..', '..', '__mocks__', 'valid-plugin'));
```

File: packages/amplify-cli/src/__tests__/plugin-helpers/verify-plugin.test.ts [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+        type: 'util',
+        commands: ['version', 'help'],
+        eventHandlers: [],
+      };
+      readJsonMock.mockReturnValueOnce(amplifyPluginJson);
+
+      // read core package.json
+      readJsonMock.mockReturnValueOnce(corePluginJson);
+
+      const result = await verifyPlugin(path.join(__dirname, '../../../__mocks__/non-event-handlers-plugin'));
```
Comment (cjihrig) [inline packages/amplify-cli/src/__tests__/plugin-helpers/verify-plugin.test.ts:317]:
```suggestion
      const result = await verifyPlugin(path.join(__dirname, '..', '..', '..', '__mocks__', 'non-event-handlers-plugin'));
```

File: packages/amplify-cli/src/__tests__/plugin-helpers/verify-plugin.test.ts [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+describe('verify-plugin', () => {
+  describe('verifyPlugin', () => {
+    beforeEach(() => {
+      const readJsonMock = JSONUtilities.readJson as jest.MockedFunction<typeof JSONUtilities.readJson>;
+      readJsonMock.mockClear();
+    });
+
+    it('returns PluginDirPathNotExist error when specify not exist path', async () => {
+      fsMock.pathExists.mockImplementation(() => Promise.resolve(false));
+      const result = await verifyPlugin('path/to/plugin');
```
Comment (cjihrig) [inline packages/amplify-cli/src/__tests__/plugin-helpers/verify-plugin.test.ts:57]: Can you make this change throughout the test file.

```suggestion
      const result = await verifyPlugin('path', 'to', 'plugin');
```

Comment (fossamagna) [reply packages/amplify-cli/src/__tests__/plugin-helpers/verify-plugin.test.ts]: Thank you for your suggestion. I will change it to use path.join function as below. 
```ts
const result = await verifyPlugin(path.join('path', 'to', 'plugin'));
```

**Review by cjihrig — APPROVED** [review]
LGTM, thanks!

**Pull request merged**